src/reactviews/pages/QueryResult/commandBar.tsx

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Button, makeStyles } from "@fluentui/react-components";
+import * as utils from "./queryResultSetup";
+import { useContext } from "react";
+import { QueryResultContext } from "./queryResultStateProvider";
+import { useVscodeWebview } from "../../common/vscodeWebviewProvider";
+import * as qr from "../../../sharedInterfaces/queryResult";
+
+const useStyles = makeStyles({
+    commandBar: {
+        display: "flex",
+        flexDirection: "column" /* Align buttons vertically */,
+    },
+    buttonImg: {
+        display: "block",
+        height: "16px",
+        width: "16px",
+    },
+});
+
+const CommandBar = () => {
+    const state = useContext(QueryResultContext);
+    const queryResultState = state?.state;
```

Comment:
if you add a check after this code, then you can avoid all the null checking later:

```
if (state === undefined) {
   return undefined;
}
```